lib/morpheus.gi

```diff
@@ -2151,7 +2151,7 @@ local m;
     IsFinite(G);
     IsFinite(H);
   fi;
-  if not IsFinite(G) and IsFinite(H) then
+  if not IsFinite(G) and not IsFinite(H) then
```

Comment:
```suggestion
  if not IsFinite(G) or not IsFinite(H) then
```

Reply:
Thinking some more about it: if one group is finite and the other is not, of course we know the answer, namely then the group is infinite... So how about this:
```gap
if IsFinite(G) <> IsFinite(H) then
  return fail;
fi;
if not IsFinite(G) then
  Error(...)
fi;
```

Reply:
I agree with @fingolfin's second suggestion here.